lib/Serialization/Deserialization.cpp

```diff
@@ -3875,9 +3884,15 @@ Expected<Type> ModuleFile::getTypeChecked(TypeID TID) {
 
     Type parentTy = getType(parentID);
 
-    // Record the type as soon as possible. Members of a nominal type often
-    // try to refer back to the type.
-    auto nominal = cast<NominalTypeDecl>(getDecl(declID));
+    auto decl = getDecl(declID);
+    if (auto *alias = dyn_cast<TypeAliasDecl>(decl)) {
+      if (alias->getAttrs().isUnavailable(ctx))
+        typeOrOffset = alias->getUnderlyingTypeLoc().getType();
+      else
+        typeOrOffset = alias->getDeclaredInterfaceType();
+      break;
+    }
+    auto nominal = cast<NominalTypeDecl>(decl);
```

Comment:
The code immediately above handles the typealias case, and tries to decide whether to preserve sugar.

(I kept doing this too: "hey, that's a `cast` I missed! oh, no, that's what I literally just handled".)